File: aspnet-in-practice/16/VB/Minifying-CSS/App_Code/CssHandler.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Configuration
Imports System.IO
Imports System.Web.UI
Imports System.Text.RegularExpressions
Imports System.Web.Caching

Namespace ASPNET4InPractice.Chapter14
    Public Class CssHandler
        Implements IHttpHandler
        Public ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
            ' true if you have no instance specific information
            Get
                Return True
            End Get
        End Property

        Public Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
            Dim fileContent As String = String.Empty
            Dim filePath As String = context.Request.PhysicalPath
            Dim cacheKey As String = String.Concat("css-", filePath)
            Dim cacheValue As Object = context.Cache(cacheKey)

            If cacheValue Is Nothing Then
                fileContent = AddInCache(filePath, cacheKey)
            Else
                fileContent = TryCast(cacheValue, String)
            End If

            ' set the appropriate content type
            context.Response.ContentType = "text/css"
            context.Response.Write(fileContent)
        End Sub

        Private Shared Function AddInCache(ByVal filePath As String, ByVal cacheKey As String) As String
            ' then write out the file
            Dim fileContent As String = File.ReadAllText(filePath)
            fileContent = String.Concat("/* minifyed at ", DateTime.Now, "*/ ", fileContent)

            fileContent = Regex.Replace(fileContent, vbTab, String.Empty)
            fileContent = Regex.Replace(fileContent, vbCr & vbLf, String.Empty)
            fileContent = Regex.Replace(fileContent, vbCr, String.Empty)
            fileContent = Regex.Replace(fileContent, vbLf, String.Empty)

            ' this will remove more than 2 spaces sequences
            fileContent = Regex.Replace(fileContent, "[ ]{2,}", String.Empty)

            ' add to cache for 2 hours
            HttpContext.Current.Cache.Insert(cacheKey, fileContent, New CacheDependency(filePath), DateTime.Now.AddHours(2), TimeSpan.Zero)

            Return fileContent
        End Function

    End Class

End Namespace

File: aspnet-in-practice/16/CS/Minifying-CSS/App_Code/CssHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Configuration;
using System.IO;
using System.Web.UI;
using System.Text.RegularExpressions;
using System.Web.Caching;

namespace ASPNET4InPractice.Chapter14
{
	public class CssHandler : IHttpHandler
	{
		public bool IsReusable
		{
			// true if you have no instance specific information
			get { return true; }
		}

		public void ProcessRequest(HttpContext context)
		{
			string fileContent = string.Empty;
			string filePath = context.Request.PhysicalPath;
			string cacheKey = string.Concat("css-", filePath);
			object cacheValue = context.Cache[cacheKey];
			
			if (cacheValue == null)
				fileContent = AddInCache(filePath, cacheKey);
			else
				fileContent = cacheValue as string;

			// set the appropriate content type
			context.Response.ContentType = "text/css";
			context.Response.Write(fileContent);
		}

		private static string AddInCache(string filePath, string cacheKey)
		{
			// then write out the file
			string fileContent = File.ReadAllText(filePath);
			fileContent = string.Concat("/* minifyed at ", DateTime.Now, "*/ ", fileContent);

			fileContent = Regex.Replace(fileContent, "\t", string.Empty);
			fileContent = Regex.Replace(fileContent, "\r\n", string.Empty);
			fileContent = Regex.Replace(fileContent, "\r", string.Empty);
			fileContent = Regex.Replace(fileContent, "\n", string.Empty);

			// this will remove more than 2 spaces sequences
			fileContent = Regex.Replace(fileContent, "[ ]{2,}", string.Empty);

			// add to cache for 2 hours
			HttpContext.Current.Cache.Insert(cacheKey, fileContent, new CacheDependency(filePath), DateTime.Now.AddHours(2), TimeSpan.Zero);

			return fileContent;
		}

	}

}